### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by jsoucheiron**
* Add release drafter config
* Bump version to 0.21.2
* Upgrade make freeze commands
* Start using ruff instead of flake8

### CODE REVIEW

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      - 'major'
+  minor:
+    labels:
+      - 'minor'
+  patch:
+    labels:
+      - 'patch'
+  default: patch
+template: |
+  ## Changes
```
Comment (jsoucheiron) [inline .github/release-drafter.yml:17]: We need to have this in master before enabling the action

File: .github/workflows/pypi-release.yml [hunk 2]
```diff
@@ -24,6 +24,6 @@ jobs:
         run: python setup.py sdist bdist_wheel
 
       - name: Publish distribution 📦 to PyPI
-        uses: pypa/gh-action-pypi-publish@master
+        uses: pypa/gh-action-pypi-publish@release/v1
```
Comment (jsoucheiron) [inline .github/workflows/pypi-release.yml:27]: It was raising a warning

File: Makefile [hunk 3]
```diff
@@ -3,11 +3,11 @@ SOURCES = $(shell find . -name "*.py")
...
 	pip install -r requirements.txt
 
-install-dev: install
-	pip install -e ".[dev]"
+install-dev:
+	pip install -r requirements.txt -r requirements-dev.txt .
 
 install-docs:
-	pip install -e ".[dev,docs]"
+	pip install -r requirements.txt -r requirements-docs.txt .
```
Comment (AliceMEd) [inline Makefile:10]: Does install docs not require the dev install anymore?

File: requirements.txt [hunk 4]
```diff
@@ -1,8 +1,8 @@
 #
-# This file is autogenerated by pip-compile with python 3.9
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.7
```
Comment (AliceMEd) [inline requirements.txt:2]: Compiled with 3.7?

Comment (jsoucheiron) [reply requirements.txt:2]: yes, it's the earliest version we support. Without this we install newer library versions that will not work with 3.7

File: Makefile (hunk 3 above)
Comment (jsoucheiron) [reply Makefile:10]: it does not

**Review by ignaciobolonio — APPROVED** [review]

**Review by marcsantamaria-sky — APPROVED** [review]

**Pull request merged**